ast/src/main/java/com/graphicsfuzz/common/ast/type/BasicType.java

```diff
@@ -456,4 +456,22 @@ public int getNumColumns() {
     assert Arrays.asList(BasicType.MAT4X2, BasicType.MAT4X3, BasicType.MAT4X4).contains(this);
     return 4;
   }
+
+  /**
+   * Finds the number of rows in a matrix type. Will cause an assertion failure if
```

Comment:
Actually how about we do this properly - make it throw an unsupported operation exception (with a message string) instead of asserting, and use a @throws clause in the JavaDoc to document that this can happen?

In general don't say "Will cause an assertion failure if" because assertions are not always enabled, so this depends on what mode you run java in (whether -ea is passed or not).

"Can only be invoked on a matrix type, and finds the number of rows of the matrix."

Reply:
But in fact I suggest that you do this as a separate PR, and do it for all the functions in the class that have that form.